src/ListFilesRequest.cpp

```diff
@@ -0,0 +1,33 @@
+#include "ListFilesRequest.h"
+
+void deleteSelfAndDie(Command * cmd, std::string s)
+{
+    perror(s.c_str());
+    delete cmd;
+    pthread_exit((void*)nullptr);
+}
+
+void ListFilesRequest::sendInfoMsgUDP(struct in_addr * in_addr)
+{
+	commonSocketAddrIn.sin_addr = *in_addr;
+	if (sendto(commonSocketFd, &requestFilesMessage, sizeof(requestFilesMessage),
+			0, (struct sockaddr*) &commonSocketAddrIn, sizeof(commonSocketAddrIn)) < 0)
+	{
+		deleteSelfAndDie(this, "sendto");
+		std::cout<<"List request to x host failed..."<<std::endl;
```

Comment:
you can print which host failed, based on struct in_addr - i suggest prints node IP address (inet_ntoa)